packages/migrate/src/migrate.ts

```diff
@@ -62,14 +62,16 @@ export default async function migrate(input: MigrateInput): Promise<MigrateOutpu
   for (const fileName of fileNames) {
     logger?.info(`Processing file: ${fileName}`);
 
+    let allChangedFiles: Set<string> = new Set();
+
     for (const pluginClass of plugins) {
       const plugin = new pluginClass(service, logger, reporter);
-      const updatedText = await plugin.run({ fileName });
-      service.setFileText(fileName, updatedText);
+      const changedFiles = await plugin.run({ fileName });
+      allChangedFiles = new Set([...allChangedFiles, ...changedFiles]);
```

Comment:
The same [as below](https://github.com/rehearsal-js/rehearsal-js/pull/176/files#r879769431), you can just use `string[]` instead of `Set`:
```
allChangedFiles = [...allChangedFiles, ...changedFiles];
```

Reply:
The benefit of using Set is that Set will automatically merge (deduplicate) same file names. The same files can be modified by the different diagnostics. If I use an array I need to check if there are file names added twice or more times into the array.

Reply:
👍